### PR TITLE
Add retrieve, update, close, and all methods to Dispute.

### PR DESCRIPTION
**Pull request by cupcait**
Also deprecate old `/v1/charges/:charge/dispute` endpoints.

r? @kyleconroy cc @jimdanz

### CODE REVIEW

File: src/test/java/com/stripe/StripeTest.java [hunk 1]
```diff
@@ -536,6 +535,80 @@ public void testDisputedCharge() throws StripeException, InterruptedException {
...
+		Dispute dispute = disputedCharge.getDispute();
+		Dispute retrievedDispute = Dispute.retrieve(dispute.getId());
+		assertEquals(dispute.getId(), retrievedDispute.getId());
+	}
+
+	@Test
+	public void testDisputeList() throws StripeException, InterruptedException {
+		createDisputedCharge(100).getDispute();
+		createDisputedCharge(150).getDispute();
+		createDisputedCharge(300).getDispute();
```
Comment (cupcait) [inline src/test/java/com/stripe/StripeTest.java]: I'm not sure that these three charge creations are necessary - there seem to be existing disputes on this account already. For now I have these here just to be extra safe, but let me know if it should be peaceful to remove them.

**Comment by cupcait** [comment]
Should I also add comments to deprecate `/charges/:charge/dispute` methods in this change?

**Comment by kyleconroy** [comment]
@cupcait In this case, we can actually use the `@Deprecated` annotation.

File: src/test/java/com/stripe/StripeTest.java (hunk 1 above)
Comment (kyleconroy) [inline src/test/java/com/stripe/StripeTest.java]: I think it's safe to remove these

**Comment by kyleconroy** [comment]
:+1:

**Pull request merged**